src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -140,6 +140,80 @@ object VariantSampleMatrix {
       tGen = (t: Type, v: Annotation) => t.genValue.resize(20),
       makeKOk = (vSig: Type) => vSig.orderedKey)
       .gen(hc)
+
+  def checkDatasetSchemasCompatible(datasets: Array[VariantSampleMatrix[Annotation, Annotation, Annotation]]) {
```

Comment:
This is a duplicate of the routine of the same name in HailContext that's used for read.

Reply:
I'm going to remove that in the next PR. I could consolidate that here, is that preferred?

Reply:
In particular, the error message is much better in the one used for read. That's why I left it (and would prefer to leave it duplicated in 0.1)